### PR TITLE
fix led panic feature

### LINKED ISSUE #10479: LED_PANIC never used?
Hello,

I tried to use the LED_PANIC feature but it is not working... Is there a logic that must be implemented to handle this case? I cannot find any call to board_autoled_on() with it in the code base.

Note: I provoke a panic with the TTY_FORCE_PANIC feature.

### PR DESCRIPTION
**Pull request by trns1997**
## Summary
- Fix LED PANIC Feature to blink at 2Hz.
- Closes #10479

### CODE REVIEW

**Comment by acassis** [comment]
Very nice @trns1997 ! Please replace the C++ comment ( // ) with C comment ( /* ..... */ )

File: sched/misc/assert.c [hunk 1]
```diff
@@ -674,6 +674,14 @@ void _assert(FAR const char *filename, int linenum,
 #else
       for (; ; )
         {
+#ifdef CONFIG_ARCH_LEDS
+          /* FLASH LEDs a 2Hz */
+          // board_autoled_on(LED_PANIC);
+          printf("ON\n");
+          up_mdelay(250);
+          printf("OFF\n");
```
Comment (acassis) [inline sched/misc/assert.c:682]: Using printf() here in the kernel doesn't appear correct

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
Please also include a Summary and commit message explaining what you want to fix exactly.

As I told you, Panic LED is working, but your board needs to include the files needed to support it. :-)

File: sched/misc/assert.c (hunk 1 above)
Comment (trns1997) [reply sched/misc/assert.c:682]: it does, i tested on the on the f411e-disco

**Comment by trns1997** [comment]
> Please also include a Summary and commit message explaining what you want to fix exactly.
> 
> As I told you, Panic LED is working, but your board needs to include the files needed to support it. :-)

This PR is clearly a WIP. I am not entirely sure what is the best course of action. Checkout the issue #10479, I think we found a problem with the LED_PANIC and LED_ASSERTION never being called.

File: sched/misc/assert.c (hunk 1 above)
Comment (xiaoxiang781216) [reply sched/misc/assert.c:682]: please polish your code.

**Comment by Siviuze** [comment]
> Please also include a Summary and commit message explaining what you want to fix exactly.
> 
> As I told you, Panic LED is working, but your board needs to include the files needed to support it. :-)

It seems that the code was lost (following the pointed PR in https://github.com/apache/nuttx/issues/10479). I cannot find the call to LED_PANIC in sched/misc/assert.c:676 where it seems to belong (cf. nuttx-11) nor anywhere in the code base (grep -rn "board_autoled_on(LED_PANIC)" return nothing).

**Comment by xiaoxiang781216** [comment]
please fix the ci compile error.

**Comment by trns1997** [comment]
> please fix the ci compile error.

@xiaoxiang781216 before i attack the fix for CI. I would like to confirm the strategy on `<arch/board/board.h>`. Because I noticed that many users have included their boards iomux/hardware/mcu specific headers in the boards `boards.h` which causes the build to fail for example:

```
   Building NuttX...
In file included from ./misc/assert.c:39:
/github/workspace/sources/nuttx/include/arch/board/board.h:34:10: fatal error: stm32_rcc.h: No such file or directory
   34 | #include "stm32_rcc.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
```

In terms of strategy should users define other mcu specific headers in this file or should they be included in sources that require it (the latter seems more logical). To make it clearer we can look at the following 2 files:

### stm32f411-minimum
file:  `boards/arm/stm32/stm32f411-minimum/include/board.h`
Include:
```
/****************************************************************************
 * Included Files
 ****************************************************************************/

#include <nuttx/config.h>
#ifndef __ASSEMBLY__
#  include <stdint.h>
#endif

#include <stm32.h> // CAUSES BUILD TO FAIL
```
In this case the build fails as it is unable to resolve `stm32.h`

### stm32f103-minimum
file: `boards/arm/stm32/stm32f103-minimum/include/board.h`
Include:
```
/****************************************************************************
 * Included Files
 ****************************************************************************/

#include <nuttx/config.h>
#ifndef __ASSEMBLY__
#  include <stdint.h>
#endif
```
Here everything works fine and the user include `stm32.h` in the sources that need it.

I would like to know your opinion. Based on that I will either go ahead and clean up the `board.h` headers for all failing boards and include them to the board sources. Otherwise i am not sure what we should do :)

**Comment by xiaoxiang781216** [comment]
@patacongo could you give some suggestion?

**Comment by gregory-nutt** [comment]
The header file issue is discussed here:  https://cwiki.apache.org/confluence/display/NUTTX/Including+Files+in+board.h including suggestions for avoiding the issue.

It was intended to work that way in order manage the scope of header files.

File: arch/arm/src/a1x/hardware/a10_piocfg.h [hunk 2]
```diff
@@ -153,12 +153,12 @@
 #define PIO_CSI1_D23    (PIO_PERIPH7 | PIO_PORT_PIOH | PIO_PIN23)
 #define PIO_CSI1_FIELD  (PIO_PERIPH7 | PIO_PORT_PIOH | PIO_PIN25)
 #define PIO_CSI1_HSYNC  (PIO_PERIPH3 | PIO_PORT_PIOG | PIO_PIN2)
-#define PIO_CSI1_HSYNC  (PIO_PERIPH7 | PIO_PORT_PIOH | PIO_PIN26)
+// #define PIO_CSI1_HSYNC  (PIO_PERIPH7 | PIO_PORT_PIOH | PIO_PIN26)
```
Comment (trns1997) [inline arch/arm/src/a1x/hardware/a10_piocfg.h:156]: Need to come back to this. The `-Werror` flag causes the pipeline to fail as there are redefinition

**Comment by TimJTi** [comment]
> Error: /home/runner/work/nuttx/nuttx/nuttx/arch/arm/src/a1x/hardware/a10_piocfg.h:120:1: error: Blank line contains whitespace

./tools/checkpatch.sh is your friend...I found out the hard way <face palm>

File: arch/arm/src/a1x/a1x_lowputc.c [hunk 3]
```diff
@@ -27,13 +27,15 @@
...
-#include <arch/board/board.h>
 
 #include "arm_internal.h"
 #include "a1x_config.h"
-#include "hardware/a1x_uart.h"
 #include "a1x_pio.h"
 
+#include "hardware/a1x_uart.h"
+
+#include <arch/board/board.h>
```
Comment (xiaoxiang781216) [inline arch/arm/src/a1x/a1x_lowputc.c:37]: why not keep it in the original place?

File: arch/arm/src/a1x/a1x_serial.c [hunk 4]
```diff
@@ -42,14 +42,16 @@
...
 #include "arm_internal.h"
 #include "chip.h"
-#include "hardware/a1x_uart.h"
+#include "a1x_config.h"
 #include "a1x_pio.h"
 #include "a1x_serial.h"
 
+#include "hardware/a1x_uart.h"
+
+#include <arch/board/board.h>
```
Comment (xiaoxiang781216) [inline arch/arm/src/a1x/a1x_serial.c:53]: ditto

File: arch/arm/src/a1x/a1x_timerisr.c [hunk 5]
```diff
@@ -31,9 +31,9 @@
 #include <nuttx/arch.h>
 
 #include <arch/irq.h>
-#include <arch/board/board.h>
 
 #include "arm_internal.h"
+
```
Comment (xiaoxiang781216) [inline arch/arm/src/a1x/a1x_timerisr.c:36]: revert

File: arch/arm/src/a1x/a1x_lowputc.c (hunk 3 above)
Comment (trns1997) [reply arch/arm/src/a1x/a1x_lowputc.c:37]: it looked more consistent, where u have nuttx specific imports first, then hardware includes then board.h. But i guess i could just revert it as it is unnecessary,

Comment (xiaoxiang781216) [reply arch/arm/src/a1x/a1x_lowputc.c:37]: normally, <> puts before "".

File: arch/arm/src/a1x/hardware/a10_piocfg.h [hunk 6]
```diff
@@ -414,10 +414,10 @@
...
 #define PIO_SDC0_D3     (PIO_PERIPH2 | PIO_PORT_PIOF | PIO_PIN4)
 
-#define PIO_SDC1_CLK    (PIO_PERIPH4 | PIO_PORT_PIOG | PIO_PIN1)
-// #define PIO_SDC1_CLK    (PIO_PERIPH5 | PIO_PORT_PIOH | PIO_PIN23)
-#define PIO_SDC1_CMD    (PIO_PERIPH4 | PIO_PORT_PIOG | PIO_PIN0)
-// #define PIO_SDC1_CMD    (PIO_PERIPH5 | PIO_PORT_PIOH | PIO_PIN22)
+#define PIO_SDC1_CLK_1  (PIO_PERIPH4 | PIO_PORT_PIOG | PIO_PIN1)
+#define PIO_SDC1_CLK_2  (PIO_PERIPH5 | PIO_PORT_PIOH | PIO_PIN23)
+#define PIO_SDC1_CMD_1  (PIO_PERIPH4 | PIO_PORT_PIOG | PIO_PIN0)
+#define PIO_SDC1_CMD_2  (PIO_PERIPH5 | PIO_PORT_PIOH | PIO_PIN22)
```
[the remark this answers is not in the document]
Comment (trns1997) [reply arch/arm/src/a1x/hardware/a10_piocfg.h:420]: i will squash all my commits soon :)

File: arch/arm/src/a1x/a1x_lowputc.c (hunk 3 above)
Comment (trns1997) [reply arch/arm/src/a1x/a1x_lowputc.c:37]: ah alright, i'll revert them then

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**